app/src/main/java/vn/mbm/phimp/me/newGallery.java

```diff
@@ -3445,7 +3448,35 @@ public void onResume()
               }catch(Exception e){
 
               }
-            
+            localPhotosGrid.setOnScrollListener(new AbsListView.OnScrollListener() {
+                @Override
+                public void onScrollStateChanged(AbsListView view, int scrollState) {
+
```

Comment:
localPhotosGrid is the view that displays images. 
I added a scroll listener to it for detecting that we have scrolled and reached to the end of the grid. When that is detected, more images gets added to the grid(this is same as manually pressing more images icon on top). 
The images will be cached for the first time so there may be a little lag(which was also present previously). Next time onwards scroll will be perfectly smooth.